vercel/turbo
# chore: revert auth json

### Description

Given current staffing and the fact that the `auth.json` isn't a big feature for the next minor release we want to get more testing to make sure that all of our existing authentication strategies still work with `auth.json`. At the moment `TURBO_TOKEN` isn't respected, this PR should fix that.
```
[0 olszewski@chriss-mbp] /Users/olszewski/code/vercel/turborepo $ TURBO_TOKEN=*** turbo_dev build --filter=docs --remote-only --output-logs new-only
 WARNING  no caches are enabled
...
```

This is a revert of the following PRs
- #6880 
- #6889 
- #6927 
which make up all of the usage of `auth.json`.

### Testing Instructions

Existing tests and 👀 making sure that I reverted everything correctly

Make sure that setting `TURBO_TOKEN` works as expected:
```
[0 olszewski@chriss-mbp] /Users/olszewski/code/vercel/turborepo $ turbo_dev info | head -n 3
You are not logged in
44 packages found in workspace

[0 olszewski@chriss-mbp] /Users/olszewski/code/vercel/turborepo $ TURBO_TOKEN=*** turbo_dev build --filter=docs --remote-only --output-logs new-only
• Packages in scope: docs
• Running build in 1 packages
• Remote caching enabled
@turbo/workspaces:build: cache hit (outputs already on disk), suppressing logs 01ead40ee62658af
docs:rss: cache hit, suppressing logs c569e716da69384d
docs:schema: cache hit, suppressing logs c7438141455c2a98
@turbo/gen:build: cache hit, suppressing logs e0ef70531aac79d7
docs:build: cache hit (outputs already on disk), suppressing logs 9e268e56e4cb8842

 Tasks:    5 successful, 5 total
Cached:    5 cached, 5 total
  Time:    1.01s >>> FULL TURBO
```

Closes TURBO-2061
